pages/api/index.ts

```diff
@@ -1,1632 +1,1704 @@
+
+        const githubAuthHeader = `token ${githubKey}`;
+        const userAgentHeader = `${repoFullName}, linear-github-sync`;
+        const issuesEndpoint = `https://api.github.com/repos/${repoFullName}/issues`;
+
+        // Map the user's Linear username to their GitHub username if not yet mapped
+        await upsertUser(
+            linear,
+            githubUserId,
+            linearUserId,
+            userAgentHeader,
+            githubAuthHeader
+        );
+
+        const syncedIssue = await prisma.syncedIssue.findFirst({
+            where: {
+                linearIssueId: data.id,
+                linearTeamId: data.teamId
+            },
+            include: { GitHubRepo: true }
+        });
+
+        if (action === "update") {
+            // Label updated on an already-Public issue
+            if (updatedFrom.labelIds?.includes(publicLabelId)) {
+                if (!syncedIssue) {
+                    console.log(skipReason("label", ticketName));
+                    return res.status(200).send({
+                        success: true,
+                        message: skipReason("label", ticketName)
+                    });
+                }
+
+                // Label(s) removed
+                if (data.labelIds.length < updatedFrom.labelIds.length) {
+                    const removedLabelId = updatedFrom.labelIds.find(
+                        id => !data.labelIds.includes(id)
+                    );
+
+                    // Public label removed
+                    if (removedLabelId === publicLabelId) {
+                        await prisma.syncedIssue.delete({
+                            where: { id: syncedIssue.id }
+                        });
+
+                        const reason = `Deleted synced issue ${ticketName} after Public label removed.`;
+                        console.log(reason);
+                        return res.status(200).send({
+                            success: true,
+                            message: reason
+                        });
+                    }
+
+                    const label = await linear.issueLabel(removedLabelId);
+                    if (!label) {
+                        return res.status(403).send({
+                            success: false,
+                            message: "Could not find label."
+                        });
+                    }
+
+                    const removedLabelResponse = await got.delete(
+                        `${GITHUB.REPO_ENDPOINT}/${syncedIssue.GitHubRepo.repoName}/issues/${syncedIssue.githubIssueNumber}/labels/${label.name}`,
+                        {
+                            headers: {
+                                Authorization: githubAuthHeader,
+                                "User-Agent": userAgentHeader
+                            }
+                        }
+                    );
+
+                    if (removedLabelResponse.statusCode > 201) {

```

Comment:
With time would suggest remove those checks and throw an error from got, and perhaps extract it to it's own class wrapper so we can reuse logic and manage headers in a single location.